fiware-rss/src/test/java/es/upm/fiware/rss/service/SettlementManagerTest.java

```diff
@@ -268,6 +266,83 @@ public void testRunSettlementAll() throws RSSException {
         verify(poolManager).closeTaskPool(callbackUrl);
     }
 
+    private SharingReport mockSharingReport(int id, boolean paid) {
+        BmCurrency curr = Mockito.mock(BmCurrency.class);
+        when(curr.getTxIso4217Code()).thenReturn("EUR");
+
+        DbeAggregator aggr = Mockito.mock(DbeAggregator.class);
+        when(aggr.getTxEmail()).thenReturn(aggregatorId);
+
+        DbeAppProviderId provId = Mockito.mock(DbeAppProviderId.class);
+        when(provId.getAggregator()).thenReturn(aggr);
+        when(provId.getTxAppProviderId()).thenReturn(providerId);
+
+        DbeAppProvider provider = Mockito.mock(DbeAppProvider.class);
+        when(provider.getId()).thenReturn(provId);
+
+        SharingReport report = Mockito.mock(SharingReport.class);
+        when(report.getId()).thenReturn(id);
+
+        when(report.getOwner()).thenReturn(provider);
+
+        when(report.getAggregatorValue()).thenReturn(new BigDecimal(0.3));
+        when(report.getAlgorithmType()).thenReturn("FIXED_PERCENTAGE");
+        when(report.getCurrency()).thenReturn(curr);
+        when(report.getOwnerValue()).thenReturn(new BigDecimal(0.7));
+        when(report.getProductClass()).thenReturn(productClass);
+        when(report.getDate()).thenReturn(new Date());
+        when(report.getPaid()).thenReturn(paid);
+        return report;
```

Comment:
In this case the object you are mocking is a been, so you can create directly the object without using mockito. Note that when you mock something you are overriding the original implementation with one that just returns the expected value, which is the same as if you make a set and then a get in a bean object.